src/assisted_installer_controller/assisted_installer_controller_test.go

```diff
@@ -1065,12 +1116,65 @@ var _ = Describe("installer HostRoleMaster role", func() {
 
 				mockbmclient.EXPECT().UpdateClusterOperator(gomock.Any(), "cluster-id", "lso", "", models.OperatorStatusFailed, "Waiting for operator timed out").Return(nil).Times(1)
 				mockbmclient.EXPECT().CompleteInstallation(gomock.Any(), "cluster-id", true, "", nil).Return(nil).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageWaitingForFinalizing).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageWaitingForClusterOperators).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageAddingRouterCa).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageWaitingForOlmOperators).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageWaitingForOlmOperatorsCsvInitialization).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageApplyingManifests).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageWaitingForOlmOperatorsCsv).Times(1)
+				mockbmclient.EXPECT().UpdateFinalizingProgress(gomock.Any(), gomock.Any(), models.FinalizingStageDone).Times(1)
+				ctx, cancel := context.WithTimeout(context.TODO(), 100*time.Millisecond)
+				defer cancel()
 
 				wg.Add(1)
-				assistedController.PostInstallConfigs(context.TODO(), &wg)
+				assistedController.PostInstallConfigs(ctx, &wg)
 				wg.Wait()
-				Expect(assistedController.Status.HasError()).Should(Equal(false))
-				Expect(assistedController.Status.GetOperatorsInError()).To(ContainElement("lso"))
+				Expect(assistedController.status.HasError()).Should(Equal(false))
+				Expect(assistedController.status.GetOperatorsInError()).To(ContainElement("lso"))
+			})
+
+			It("waiting for single OLM operator which soft timeouts", func() {
```

Comment:
which -> with?

Reply:
Failure in OLM operators bring cluster to degraded state and not error, so why need soft timeouts on it?

Reply:
Maybe the title was misleading